src/overlays/actors/ovl_Obj_Jgame_Light/z_obj_jgame_light.c

```diff
@@ -28,29 +33,173 @@ const ActorInit Obj_Jgame_Light_InitVars = {
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    Lights_PointGlowSetInfo(lights, this->actor.world.pos.x, (this->actor.world.pos.y + 70.0f), this->actor.world.pos.z,
+                            0xFF, 0xFF, 0xB4, -1);
+    this->lightNode = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->lightInfo);
+    Actor_SetFocus(&this->actor, 60.0f);
+    this->actor.colChkInfo.health = 0;
+    this->unk_1B8 = 0;
+    this->unk_1B6 = 0;
+    this->unk_1AC = 0;
+    this->unk_1B2 = 0;
+    this->unk_1B4 = 0;
+    this->unk_1A8 = 0.0f;
+}
+
+void ObjJgameLight_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjJgameLight* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    LightContext_RemoveLight(globalCtx, &globalCtx->lightCtx, this->lightNode);
+}
+
+void func_80C15474(ObjJgameLight* this, GlobalContext* globalCtx) {
+    u8 temp_a1;
+
+    if (((this->actor.colChkInfo.health & 1) != 0) && (this->unk_1B6 == 0)) {
+        if (this->unk_1AC < 0xA0) {
+            this->unk_1AC += 0x28;
+        } else {
+            this->unk_1AC = 0xC8;
+            this->unk_1B6 = 1;
+        }
+        if (this->unk_1A8 < 0.7f) {
+            this->unk_1A8 += 0.3f;
+        } else {
+            this->unk_1A8 = 1.0f;
+        }
+    } else if ((this->actor.colChkInfo.health & 8) != 0) {
+        if (this->unk_1AC >= 0x29) {
+            this->unk_1AC -= 0x28;
+        } else {
+            this->unk_1AC = -1;
+            if (this->unk_1A8 == 0.0f) {
+                this->unk_1B6 = 0;
+                this->actor.colChkInfo.health &= 0xFFFE;
+                this->actor.colChkInfo.health &= (s16)0xFFF7;
+            }
+        }
+        if (this->unk_1A8 > 0.3f) {
+            this->unk_1A8 -= 0.3f;
+        } else {
+            this->unk_1A8 = 0.0f;
+        }
+    }
+    if (this->unk_1A8 > 0.1f) {
+        func_800B9010(&this->actor, NA_SE_EV_TORCH - SFX_FLAG);
+    }
+    temp_a1 = (((s32)(Rand_ZeroOne() * 127.0f) + 0x80));
+    Lights_PointSetColorAndRadius(&this->lightInfo, temp_a1, (u32)(temp_a1 * 0.7f) & 0xFF, 0, this->unk_1AC);
+}
+
+void func_80C156C4(ObjJgameLight* this, GlobalContext* globalCtx) {
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+void func_80C15718(ObjJgameLight* this, GlobalContext* globalCtx) {
+    u8 colChk;
+
+    if (((this->actor.colChkInfo.health & 1) != 0) && (((this->unk_1B8) & 1) == 0)) {
+        Audio_PlaySfxAtPos(&this->actor.projectedPos, NA_SE_EV_FLAME_IGNITION);
+        this->unk_1B8 = colChk = this->actor.colChkInfo.health;
+    }
+    if ((this->actor.colChkInfo.health & 2) != 0) {
+        this->actor.colChkInfo.health = this->actor.colChkInfo.health & 0xFFFD;
```

Comment:
Throughout the health is being used as a set of flags, can remove the `!= 0`s in conditionals and convert these `& 0xFFFx`s into `~`s. And make a set of defines for the flags, although check the Fisherman's C file to see if it needs to reference them.